File: src/OnlineCoursesPlatform.Application/Features/Courses/Dto/CourseDto.cs
namespace OnlineCoursesPlatform.Application.Features.Courses.Dto
{
    public class CourseDto
    {
        public int Id { get; set; }
        public string Title { get; set; } = null!;
        public string Description { get; set; } = null!;
        public int? CategoryId { get; set; }
        public string? CategoryName { get; set; }
        public DateTime DateCreated { get; set; }
        public DateTime DateModified { get; set; }
        public string Difficulty { get; set; } = null!;

        public int LessonsCount { get; set; }
        public string? AuthorAvatarUrl { get; set; }
        public int AuthorId { get; set; }
        public string? About { get; set; }
    }

}

File: src/OnlineCoursesPlatform.Frontend/src/interfaces/CourseDto.ts
export interface CreateCourseRequest {
  title: string;
  description: string;
  categoryId: number;
  tagIds: number[];
  difficulty: "Beginner" | "Intermediate" | "Advanced"; 
  about?: string | null;
}

export interface TeacherOwnCourse {
  id: number;
  title: string;
  description: string;
  dateCreated: string;
  dateModified: string;
   difficulty: "Beginner" | "Intermediate" | "Advanced";
  categoryName: string;
  lessonsCount: number;
  authorAvatarUrl: string | null;
  authorId: number;
  about?: string | null;
}

export interface PublicCourse {
  id: number;
  title: string;
  description: string;
  dateCreated: string;
  dateModified: string;
  difficulty: "Beginner" | "Intermediate" | "Advanced";
  categoryName: string;
  lessonsCount: number;
  authorAvatarUrl: string | null;
  authorId: number;
  about?: string | null;
}

export interface CourseDetails {
  id: number;
  title: string;
  description: string;
  categoryId: number | null;
  dateCreated: string;
  dateModified: string;
  authorId: number;
  about?: string | null;
   categoryName: string;
  difficulty: "Beginner" | "Intermediate" | "Advanced";
}